cosmwasm/its.ts

```diff
@@ -0,0 +1,186 @@
+    const supply = await token.totalSupply();
+    return supply.toString();
+}
+
+// TODO tkulik: This command will be used in the supply alignment command.
+export async function isTokenSupplyTracked(tokenManagerType: number, token: Contract): Promise<boolean> {
+    return tokenManagerType === tokenManagerTypes.NATIVE_INTERCHAIN_TOKEN && (await token.isMinter(constants.AddressZero));
+}
+
+function formatTokenId(tokenAddress: string): string {
+    if (tokenAddress.startsWith('0x')) {
+        return tokenAddress.slice(2);
+    }
+    return tokenAddress;
+}
+
+async function registerP2pToken(client: ClientManager, config: ConfigManager, options) {
+    const { chain, tokenId, originChain, decimals, dryRun } = options;
+    try {
+        const tokenData = {
+            tokenId: tokenId,
+            originChain: originChain,
+            decimals: decimals,
+            chainName: chain,
+        };
+        const interchainTokenServiceAddress = config.getContractConfig('InterchainTokenService').address;
+        validateParameters({
+            isNonEmptyString: { interchainTokenServiceAddress },
+        });
+
+        await registerToken(config, interchainTokenServiceAddress, client, tokenData, dryRun);
+    } catch (e) {
+        printError(`Error registering token ${tokenId} on ${chain}: ${e}`);
+    }
+}
+
+async function checkTokenRegistration(client: ClientManager, config: ConfigManager, options) {
+    const { tokenId } = options;
+
+    const interchainTokenServiceAddress = config.getContractConfig('InterchainTokenService').address;
+    validateParameters({
+        isNonEmptyString: { interchainTokenServiceAddress },
+    });
+
+    const registeredChains = (
+        await Promise.all(
+            Object.keys(config.chains).map(async (axelarChainId: string) => {
+                const registered = await checkSingleTokenRegistration(
+                    config,
+                    client,
+                    interchainTokenServiceAddress,
+                    tokenId,
+                    axelarChainId,
+                );
+                if (registered) {
+                    return axelarChainId;
+                }
+            }),
+        )
+    ).filter(Boolean);
+
+    if (registeredChains.length === 0) {
+        printInfo(`Token ${tokenId} is not registered on any chain`);
+        return;
+    }
+
+    printInfo(`Token ${tokenId} is registered on: ${registeredChains.join(', ')}`);
+}
+
+const programHandler = () => {
+    const program = new Command();
+
+    program
+        .name('ITS p2p token registration')
+        .description('Script to perform ITS p2p token registration and check tokens registration status.');
+
+    const registerP2pTokenCmd = program
+        .command('register-p2p-token')
+        .description('Register a single P2P consensus token to the ITS Hub.')
+        .addOption(new Option('--chain <chain>', 'axelar chain id to run the script for').env('CHAIN').makeOptionMandatory(true))
```

Comment:
No need to add `env var` option for every flag, it's only useful for a few things, otherwise CLI args are best